activity_calendar/forms.py

```diff
@@ -238,15 +243,28 @@ class RegisterNewSlotForm(RegisterAcitivityMixin, ModelForm):
 
     class Meta:
         model = ActivitySlot
-        fields = ['title', 'description', 'location', 'max_participants']
+        fields = ['title', 'description', 'location', 'image', 'max_participants']
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if not self.user.has_perm('achievements.can_select_slot_image'):
+            # User does not have the required permissions to select an alternative slot image
+            # Remove the field
+            del self.fields['image']
+        else:
+            # Get the PrestImages available to the user
+            self.fields['image'].queryset = PresetImage.objects.for_user(self.user)
+
 
     def check_validity(self, data):
         super(RegisterNewSlotForm, self).check_validity(data)
 
         # Is user allowed to create a slot
-        if self.activity.slot_creation == "CREATION_NONE" and self.user.is_staff:
+        if self.activity.slot_creation == "CREATION_USER":
             pass
-        elif self.activity.slot_creation == "CREATION_USER":
+        elif self.activity.slot_creation == "CREATION_NONE" \
```

Comment:
I was indeed planning on updating from master because of the changes with the activitymoment (that are currently not on this branch yet). This and the relatively low coverage are reasons why this is still marked as a draft